src/lib/ssg.tsx

```diff
@@ -1,122 +1,94 @@
-import fs from "fs";
-import { sync } from "glob";
-import { join, relative } from "path";
-import matter from "gray-matter";
-
 import { createClient, Entry } from "contentful";
-import { Document } from '@contentful/rich-text-types';
+import { Document } from "@contentful/rich-text-types";
 
 export interface DocMeta {
   title: string;
   chapter: string;
   subchapter: string;
   section: string;
   content?: string;
-  slug?: string | string[];
-  post?: Document
+  slug?: string;
+  post?: Document;
 }
 
 interface slugRetrieved {
-  slug: string
+  slug: string;
 }
 
-// Queries the list of slugs
-export const getDocSlugs = (subDir: string): string[] => {
-  const docsDirectory = join(process.cwd(), `docs/${subDir}`);
-  const files = sync(join(docsDirectory, "**/*.md"));
-  const fileNames = files.map((file) => relative(docsDirectory, file));
-  return fileNames;
-};
-
 export async function getDocSlugsFromCMS(): Promise<string[]> {
   const client = createClient({
     space: process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID,
     accessToken: process.env.NEXT_PUBLIC_CONTENTFUL_ACCESS_KEY,
   });
 
   const res = await client.getEntries<slugRetrieved>({
-    content_type: "admissions", 
-    select: "fields.slug"
-  })
+    content_type: "admissions",
+    select: "fields.slug",
+  });
 
-  return res.items.map((x: Entry<slugRetrieved>) => x.fields.slug);
-};
+  return res.items.map((x: Entry<slugRetrieved>) => {
+    return x.fields.slug;
+  });
+}
 
-export async function getDocBySlugFromCMS(slug: string | string[]): Promise<DocMeta> {
+export async function getDocsBySlugsFromCMS(
+  slug: string | string[]
+): Promise<DocMeta[]> {
   const client = createClient({
     space: process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID,
     accessToken: process.env.NEXT_PUBLIC_CONTENTFUL_ACCESS_KEY,
   });
 
   if (Array.isArray(slug)) {
-    slug = slug.join('-');
+    slug = slug.join(",");
   }
+
+  // Less than ideal solution, since it will match with before-01, etc
+  // However, this would save the entries fetch time to 1 API call
   const res = await client.getEntries<DocMeta>({
-    content_type: "admissions", 
-    'fields.slug[match]': slug
-  })
+    content_type: "admissions",
+    "fields.slug[in]": slug,
+  });
 
-  // Obviously wrong + potentially buggy implementation since match may return more than one doc
-  // To be dealt with some other time
-  return res.items[0].fields;
+  console.debug(`Getting response docs items for : ${res.items}`);
+
+  return res.items.map((o) => o.fields);
 }
```

Comment:
Sama ini kenapa g make use of getDocBySlug ya? Ato d blh getDocBySlug yg pake ini? Soalnya basically mreka sama right

Reply:
sama aja, cuma kalo lets say kita ad 20 articles, kita make 20 rest API, kayak 1 rest roughly 200 ms aj udah makan 4s, ganggu aj

Reply:
G slh okk